packages/back-end/src/util/mongo.util.ts

```diff
@@ -147,7 +146,11 @@ export function removeMongooseFields<T>(
   if (doc.toJSON) {
     doc = doc.toJSON({ flattenMaps: true });
   }
-  return omit(doc, ["_id", "__v"]) as T;
+
+  // Delete mongoose fields rather than using lodash.omit for perf reasons since this is called a lot
+  delete doc._id;
+  delete doc.__v;
```

Comment:
The usual concern with modifying the object in place is value sharing.

Are we sure that `doc` is never used concurrently in other places? For instance, can this mess up a concurrent `update` or `find` call that expects the `doc` object to also have those elements present?

For having been in such position before, tracking down these type of imperative issues is very hard to do so, usually, the performance gains have to be weighted against the risk of falling into such a situation.

Reply:
Yes that is the usual concern.  As best as I can tell we don't use _id and __v ourselves.  I only see one reference in AuthRefreshModel and we do not run a `toInterface` function on it before then updating.  As long as we are not updating by `_id` afterwards (usually we use `id` instead), then I *think* we are safe, but I'm not positive.

Reply:
I've updated this to now copy with the spread operator, so it won't be mutating.  It doesn't save as much as before, but still might be saving something like 12%.